src/__template__/ComponentTemplate.vue

```diff
@@ -98,6 +98,7 @@ const rootElementStyles = computed((): Record<string, string> => ({
   height: iconSize.value,
   lineHeight: '0',
   width: iconSize.value,
+  pointerEvents: props.decorative ? 'none' : 'all',
```

Comment:
Should we make it automatic, or just map it to a prop instead similar to `display`? 

I'm trying to think if there's a use case whereby a decorative icon should still be able to be clicked

Reply:
When `decorative` is true we're setting `aria-hidden="true"` attribute which makes the element pretty much invisible to assistive technologies, therefore inaccessible. IMO what's inaccessible for assistive technologies shouldn't be clickable either? If this logic is correct then yes, this can and should be automatic.

Reply:
Sounds good